lib/handlers/subscribe.js

```diff
@@ -60,6 +60,21 @@ function storeSubscriptions (sub, done) {
   var broker = client.broker
   var perst = broker.persistence
 
+  if (sub && packet.subscriptions[0].topic !== sub.topic) {
+    // don't call addSubscriptions per topic,
+    // TODO change aedes subscribe handle, but this is a fast & dirty fix for now
+    return done(null, sub)
+  }
+
+  if (!packet.messageId) {
+    // this is a restoreSubscription, don't addSubscriptions!
+    // TODO change aedes subscribe handle, but this is a fast & dirty fix for now
+    if (perst._addedSubscriptions) {
+      perst._addedSubscriptions(client, packet.subscriptions)
+    }
```

Comment:
The event will not work, as https://github.com/mcollina/aedes-cached-persistence/blob/master/index.js#L79-L107 needs a callback.

BTW, are you sure you need to call `_addedSubscriptions` at all? https://github.com/mcollina/aedes-cached-persistence/blob/master/index.js#L79-L107 is used to update the cache in the other peers, but as it is a restore, I do not think we need it.